File: leedcode/moveZero/3.js
var moveZeroes = function(nums) {
    //k标记0的位置
    let k = 0
    for(let i = 0 ; i < nums.length ; i++){
        if(nums[i]){
            if(k != i){
                //交换位置
                nums[k] = nums[i]
                nums[i] = 0
                k = i
            }
            else{
                k++
            }
        }
    }return nums
};

File: leedcode/two_sum/3.go
// js 动态语言
// twoSum undefinded => functon
// go 类型约束 现代的C语言
func twoSum(nums []int, target int) []int  {
	//O(n^2) => O(n)
	// json 7
	//hashTable {7:0,}
	h := make(map[int]int) // hashTalbe
	// 
	for i,value := range nums {
		if wanted, ok := h[value] ; ok {
			return []int{wanted, i}
		}else {
			h[target-value] = i
		}
	}	
	return nil
}